pkg/pool-weighted/contracts/lbp/LiquidityBootstrappingPool.sol

```diff
@@ -123,166 +61,96 @@ contract LiquidityBootstrappingPool is IMinimalSwapInfoPool, NewBasePool {
      */
-    function updateWeightsGradually(
-        uint256 startTime,
-        uint256 endTime,
-        uint256[] memory endWeights
-    ) external authenticate whenNotPaused {
-        InputHelpers.ensureInputLengthMatch(_getTotalTokens(), endWeights.length);
-
-        startTime = GradualValueChange.resolveStartTime(startTime, endTime);
-        _startGradualWeightChange(startTime, endTime, _getNormalizedWeights(), endWeights);
-    }
-
-    // Internal functions
-
-    function _getNormalizedWeight(IERC20 token) internal view returns (uint256) {
-        uint256 i;
-
-        // First, convert token address to a token index
-
-        // prettier-ignore
-        if (token == _token0) { i = 0; }
-        else if (token == _token1) { i = 1; }
-        else if (token == _token2) { i = 2; }
-        else if (token == _token3) { i = 3; }
-        else {
-            _revert(Errors.INVALID_TOKEN);
-        }
-
-        return _getNormalizedWeightByIndex(i, _poolState);
-    }
-
-    function _getNormalizedWeightByIndex(uint256 i, bytes32 poolState) internal view returns (uint256) {
-        uint256 startWeight = poolState
-            .decodeUint(_START_WEIGHT_OFFSET + i * _START_WEIGHT_BIT_LENGTH, _START_WEIGHT_BIT_LENGTH)
-            .decompress(_START_WEIGHT_BIT_LENGTH);
-        uint256 endWeight = poolState
-            .decodeUint(_END_WEIGHT_OFFSET + i * _END_WEIGHT_BIT_LENGTH, _END_WEIGHT_BIT_LENGTH)
-            .decompress(_END_WEIGHT_BIT_LENGTH);
-        uint256 startTime = poolState.decodeUint(_START_TIME_OFFSET, _TIMESTAMP_BIT_LENGTH);
-        uint256 endTime = poolState.decodeUint(_END_TIME_OFFSET, _TIMESTAMP_BIT_LENGTH);
-
-        return GradualValueChange.getInterpolatedValue(startWeight, endWeight, startTime, endTime);
-    }
-
-    function _getNormalizedWeights() internal view returns (uint256[] memory) {
-        uint256 totalTokens = _getTotalTokens();
-        uint256[] memory normalizedWeights = new uint256[](totalTokens);
-
-        bytes32 poolState = _poolState;
-
-        // prettier-ignore
-        {
-            normalizedWeights[0] = _getNormalizedWeightByIndex(0, poolState);
-            normalizedWeights[1] = _getNormalizedWeightByIndex(1, poolState);
-            if (totalTokens == 2) return normalizedWeights;
-            normalizedWeights[2] = _getNormalizedWeightByIndex(2, poolState);
-            if (totalTokens == 3) return normalizedWeights;
-            normalizedWeights[3] = _getNormalizedWeightByIndex(3, poolState);
-        }
-
-        return normalizedWeights;
+    function _subtractSwapFeeAmount(uint256 amount) internal view returns (uint256) {
+        // This returns amount - fee amount, so we round up (favoring a higher fee amount).
+        uint256 feeAmount = amount.mulUp(getSwapFeePercentage());
+        return amount.sub(feeAmount);
     }
 
-    // Pool callback functions
-
     // Prevent any account other than the owner from joining the pool
```

Comment:
```suggestion
```
This comment was here before, but it's describing an implementation detail of the function, vs what the function does - and the same comment appears later, where it should be - so I don't think we need this here.

Reply:
Also, I can't reach this in the diff, but I would also reword this:

```
// Set the initial BPT to the value of the invariant times the number of tokens. This makes BPT supply more
// consistent in Pools with similar compositions but different number of tokens.
```

To:
```
// Set the initial BPT to the value of the invariant times the number of tokens. This makes the BPT supply
// more consistent in Pools with similar token composition, but a different number of tokens.
```